src/ua_securechannel.c

```diff
@@ -610,11 +610,9 @@ UA_SecureChannel_sendSymmetricMessage(UA_SecureChannel *channel, UA_UInt32 reque
 
     /* Encoding failed, release the message */
     if(retval != UA_STATUSCODE_GOOD) {
-        if(!ci.final) {
-            /* the abort message was not sent */
-            ci.errorCode = retval;
+        /* the abort message was not sent */
+        if(!ci.final)
             sendChunkSymmetric(&ci, &buf_start, &buf_end);
```

Comment:
Shouldn't we replace the message that didn't get encoded correctly with the abort message body before sending something? The status code from sendChunkSymmetric isn't checked here as well. Maybe log it if we don't want to override the other statuscode?

Reply:
I added a TODO with first ideas on error handling.